sdks/java/core/src/test/java/org/apache/beam/sdk/testing/ExpectedLogs.java

```diff
@@ -301,5 +302,9 @@ public void flush() {}
 
     @Override
     public void close() throws SecurityException {}
+
+    public void reset() {
+      logRecords = new ConcurrentLinkedDeque<>();
```

Comment:
make logRecords member final and just call clear on it, also make getLogs and reset private since they should only be used from the context of ExpectedLogs.